[nativert] Expose ModelRunner to public through pmpl type ModelRunnerHandle.

zhxchen17 opened the pull request:
Summary:
Today users outside of pytorch core cannot `#include <torch/nativert/ModelRunner.h>`.

It turns out that we should place a header inside `torch/csrc/api/include/`. Placing every single nativert header here would pollute the namespace a lot and that's not what we want in general. Therefore here we just create a Handle type which hold a pointer to decouple the actual type from header definition.

Test Plan:
CI

Rollback Plan:

Differential Revision: D79751098

JacobSzwejbka commented on torch/csrc/api/include/torch/nativert/ModelRunnerHandle.h:
```diff
@@ -0,0 +1,46 @@
+// incomplete ModelRunner here.
+class ModelRunner;
+
+class TORCH_API ModelRunnerHandle {
```
So going to have to add every function we end up needing from nativeRT in ET desktop to these handls over time?

zhxchen17 replied:
yeah for now. hopefully there should be <10 of them

zhxchen17 committed:
[nativert] Expose ModelRunner to public through pmpl type ModelRunnerHandle. (#159989)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/159989

Today users outside of pytorch core cannot `#include <torch/nativert/ModelRunner.h>`.

It turns out that we should place a header inside `torch/csrc/api/include/`. Placing every single nativert header here would pollute the namespace a lot and that's not what we want in general. Therefore here we just create a Handle type which hold a pointer to decouple the actual type from header definition.

Test Plan:
CI

Rollback Plan:

Differential Revision: D79751098
